pkg/controllers/status/cluster_status_controller.go

```diff
@@ -231,11 +232,16 @@ func (c *ClusterStatusController) setCurrentClusterStatus(clusterClient *util.Cl
 
 	// get the list of APIs installed in the member cluster
 	apiEnables, err := getAPIEnablements(clusterClient)
-	if len(apiEnables) == 0 {
-		klog.Errorf("Failed to get any APIs installed in Cluster %s. Error: %v.", cluster.GetName(), err)
-	} else if err != nil {
-		klog.Warningf("Maybe get partial(%d) APIs installed in Cluster %s. Error: %v.", len(apiEnables), cluster.GetName(), err)
+	if err != nil {
+		klog.Errorf("Failed to get APIs installed in Cluster %s. Error: %v.", cluster.GetName(), err)
+		return err
```

Comment:
I am a little worried that this may cause other problems. If an error is returned here, the cluster will be unhealthy. If some APIs cannot be obtained, the cluster will be unhealthy.

Reply:
would it be unhealthy ? seems the cluster needs to be healthy&online to reach getAPIEnablements, if err returned the reconcile will be requeue.

Reply:
When changing from an unhealthy state to a healthy state, if some APIs have problems, an error will be returned here and the queue will be requeued. Unable to update the cluster to a healthy state

Reply:
sounds reasonable, maybe it's better keep apiEnablements unchanged when err happens.

Reply:
no for above, if we do that the incorrect status will be kept until next reconcile which is not good at all